### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by sjatkins12**

### CODE REVIEW

File: input_parser/quoting.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	if (c == '\'')
+		return (SINGLE_QUOTE);
+	else if (c == '\"')
+		return (DOUBLE_QUOTE);
+	else if (c == '\\')
+		return (BACKSLASH);
+	return (0);
+}
+
+int				handle_embedded_quotes(uint8_t *quoted,
```
Comment (suedadam) [inline input_parser/quoting.c]: I'll leave this up to you, but I'd suggest to make this function a distributor function to call other functions depending on the `if/elseif` statements. That way it norms.

File: input_parser/parser.c [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+{
+	t_parser		*par;
+	int				ret;
+
+	if (!(par = init_parser()))
+		return (NULL);
+	while (*input_str)
+	{
+		ret = UNUSED_CHAR;
+		if (!(ret = is_semi(par, *input_str)) ||
```
Comment (suedadam) [inline input_parser/parser.c]: This is too long for the function (exceeds norm) - There's 2 real ways you can fix it at this point. I'll let you pick your poison.

- Dispatch table
- Inline function to call each and return the result

**Review by suedadam — CHANGES_REQUESTED** [review]
Please fix the last two things and then you should be good! I'll deal with the rest. Please also point out which files are needed to be merged as I'll be merging based off of specific files instead of the whole branch :)

**Comment by tle-huu** [comment]
Norm issues remaining

**Review by suedadam — APPROVED** [review]

**Pull request merged**